### PR TITLE
removing all enemies from gamearea + entity service on level change

### PR DESCRIPTION
**Pull request by yourlow**
Clearing all entites from game area entities list and entity service list to fix segfault bug

### CODE REVIEW

**Comment by yourlow** [comment]
This 'solution' also brings up the question of player state, if we remove all entites from the gameworld this means the players items + health will be lost. A couple fixes to this could be, either we could not destroy the player during this phase so they stay on the entity list, or we could make the player state global and just 'save' the state on level leave, remove all entites, and 'load' the players state back into a new player entity on level load.

**Comment by 25Ericcheong** [comment]
I looked at your code and I don't think this actually solves the problem. It might be code duplicate even. What you've included does what `entity.dispose()` does and when `gameArea.dispose()` is called, all entities would call their `dispose()` function which does exactly what your method does - it disposes all components first and then unregisters itself from the entity service.

File: source/core/src/main/com/deco2800/game/areas/GameArea.java [hunk 1]
```diff
@@ -30,9 +30,19 @@ protected GameArea() {
...
   /** Dispose of all internal entities in the area */
   public void dispose() {
+
     for (Entity entity : areaEntities) {
       entity.dispose();
     }
+
+  }
+  /** Removes all entites from the entity service and the gameArea */
+  public void removeEntites() {
```
Comment (25Ericcheong) [inline source/core/src/main/com/deco2800/game/areas/GameArea.java:40]: entity.dispose() does this already.

**Review by 25Ericcheong — COMMENTED** [review]
image
Same error even with this fix. For player state, we were planning on having another json / config file which we'll edit when player reaches the end of the level and as a new level is created - that file will be loaded to load player to its current and correct state with all the relevant data in tact if that makes sense.

**Review by Bluehalomango — COMMENTED** [review]
I've had a look at your stuff and implemented some of it into a fix of my own for items. Similar to what Eric said, the only addition you've noticeably added is the areaEntities.clear() which I added to my fix. Once we push our stuff later tonight, you can either make changes or additions for your fix. The main thing I added was the gameArea to ServiceLocator so that you can call despawnEntity instead of disposing of an entity. The issues I've found so far are from not removing entities from areaEntities after disposing of it, which despawnEntity gets around.

**Pull request closed without merging**